app/models/event.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+class Event
+  include ActiveModel::Model
+
+  attr_accessor :labware, :audit
+
+  validates :labware, :audit, presence: true
+
+  validate :check_location_exists
+
+  delegate :coordinate, to: :labware, allow_nil: true
+  delegate :location, to: :labware
+
+  def uuid
+    @uuid ||= audit.uuid
+  end
+
+  def timestamp
+    @timestamp ||= Time.zone.now
+  end
```

Comment:
Just thought, we should probably get this from audit.created_at rather than 'now', because in the case where we are regenerating Events from audits, we want the timestamp 'occured_at' to be when the event actually happened, not when the code is run.

Reply:
yes true